snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```diff
@@ -0,0 +1,237 @@
+        CatalogUtil.loadFileIO(fileIOImpl, properties, conf),
+        properties);
+  }
+
+  /**
+   * Initialize using caller-supplied SnowflakeClient and FileIO.
+   *
+   * @param name The name of the catalog, defaults to "snowflake_catalog"
+   * @param snowflakeClient The client encapsulating network communication with Snowflake
+   * @param fileIO The {@link FileIO} to use for table operations
+   * @param properties The catalog options to use and propagate to dependencies
+   */
+  @SuppressWarnings("checkstyle:HiddenField")
+  public void initialize(
+      String name, SnowflakeClient snowflakeClient, FileIO fileIO, Map<String, String> properties) {
+    Preconditions.checkArgument(null != snowflakeClient, "snowflakeClient must be non-null");
+    Preconditions.checkArgument(null != fileIO, "fileIO must be non-null");
+    this.catalogName = name == null ? DEFAULT_CATALOG_NAME : name;
+    this.snowflakeClient = snowflakeClient;
+    this.fileIO = fileIO;
+    this.catalogProperties = properties;
+    this.closeableGroup = new CloseableGroup();
+    closeableGroup.addCloseable(snowflakeClient);
+    closeableGroup.addCloseable(fileIO);
+    closeableGroup.setSuppressCloseFailure(true);
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (null != closeableGroup) {
+      closeableGroup.close();
+    }
+  }
+
+  @Override
+  public void createNamespace(Namespace namespace, Map<String, String> metadata) {
+    throw new UnsupportedOperationException(
+        String.format("createNamespace not supported; attempted for namespace '%s'", namespace));
+  }
+
+  @Override
+  public List<Namespace> listNamespaces(Namespace namespace) {
+    LOG.debug("listNamespaces with namespace: {}", namespace);
+    SnowflakeIdentifier scope = NamespaceHelpers.getSnowflakeIdentifierForNamespace(namespace);
+    Preconditions.checkArgument(
+        scope.getType() == SnowflakeIdentifier.Type.ROOT
+            || scope.getType() == SnowflakeIdentifier.Type.DATABASE,
+        "listNamespaces must be at either ROOT or DATABASE level; got %s from namespace %s",
+        scope,
+        namespace);
+    List<SnowflakeIdentifier> sfSchemas = snowflakeClient.listSchemas(scope);
+
+    List<Namespace> namespaceList =
+        sfSchemas.stream()
+            .map(
+                schema -> {
+                  Preconditions.checkState(
+                      schema.getType() == SnowflakeIdentifier.Type.SCHEMA,
+                      "Got identifier of type %s from listSchemas for %s",
+                      schema.getType(),
+                      namespace);
+                  return Namespace.of(schema.getDatabaseName(), schema.getSchemaName());
+                })
+            .collect(Collectors.toList());
+    return namespaceList;
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(Namespace namespace)
+      throws NoSuchNamespaceException {
+    LOG.debug("loadNamespaceMetadata with namespace: {}", namespace);
```

Comment:
We don't typically debug individual calls like this. What is the value here?

Reply:
Mostly have found method-call level debug logs useful when a top-level call results in multiple possibly non-obvious underlying calls (e.g. when reading a table goes through newTableOps -> doRefresh, etc) but I agree these top-level calls that mostly correspond one-to-one with user commands probably don't benefit much from these. Removed from this class.